test/unittest/tests/block/test_block_api_write.cpp

```diff
@@ -0,0 +1,279 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_api]: block_write.c
+ * The block manager performs file operations on specific blocks which eventually gets translated
+ * onto files. This test suite aims to test the possible cases for the write(), read()
+ * and write_size() API operations.
+ */
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <filesystem>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include "../wrappers/item_wrapper.h"
+
+const int DEFAULT_BLOCK_SIZE = 256;
+const std::string ALLOCATION_SIZE = "256";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+typedef std::pair<std::array<uint8_t, WT_BTREE_MAX_ADDR_COOKIE>, size_t> addr_cookie;
```

Comment:
I would rather we define a struct instead of using std::pair. Something like:
stuct cookie {
    uint8_t[MAX_COOKIE_LEN] cookie;
    size_t size;
}